js/game.js

```diff
@@ -0,0 +1,256 @@
+    // context.fillStyle = BASE.CANVAS_TEXTCOLOR;
+    // context.font = BASE.CANVAS_FONT;
+    // context.textBaseline = 'top';
+    // context.fillText ("FRAG", 50, 90);
+    // context.fillText ("Shoot to kill!!", 50, 200);
+    if (!currentMenu) {
+      playerOneMenu = new Menu();
+      currentMenu = playerOneMenu;
+      // this allows for the ship selection to be entered,
+      // as well as the switch to PlayerTwo's menu
+      bindKeyGameMenu();
+    }
+    currentMenu.render(context);
+  }
+
+  function gameStateInit() {
+    if (!gameInitialized) {
+      var P1 = utils.playerOneChoice, P2 = utils.playerTwoChoice;
+
+      context.fillStyle = BASE.CANVAS_BACKGROUND;
+      context.fillRect(0, 0, BASE.CANVAS_WIDTH, BASE.CANVAS_HEIGHT);
+      context.fillStyle = '#ffffff';
+      context.font = BASE.CANVAS_FONT;
+      context.textBaseline = 'top';
+      context.fillText ("LOADING", 200, 200);
+
+      // initialize sounds object pools using AudioFX
+      // https://github.com/jakesgordon/javascript-audio-fx
+      playerOneSound = AudioFX('resources/audio/shot' + P1 + '.wav', { volume: 1, pool: 10 });
+      playerTwoSound = AudioFX('resources/audio/shot' + P2 + '.wav', { volume: 1, pool: 10 });
+      impactSound = AudioFX('resources/audio/impact.wav', { volume: 1, pool: 10 });
+
+      // initialize ship objects
+      playerOne = new Ship(START_POS_ONE, utils.shipConfig[P1], playerOneSound);
+      playerTwo = new Ship(START_POS_TWO, utils.shipConfig[P2], playerTwoSound);
+
+      // initialize player lives
+      playerOneLife = new Life(1);
+      playerTwoLife = new Life(2);
+
+      // bind game keys
+      utils.bindKeys(playerOne, playerTwo);
+      // bind enter key to enable restart after game end
+      bindKeyRestart(switchGameState);
+      gameInitialized = true;
+      // hack to remove flicker between game state transitions
+      setTimeout(function() {
+        switchGameState(GAME_STATE_PLAY);
+      },2000);
+    }
+  }
+
+  function gameStatePlay() {
+    context.fillStyle = BASE.CANVAS_BACKGROUND;
+    context.fillRect(0, 0, BASE.CANVAS_WIDTH, BASE.CANVAS_HEIGHT);
+    playerOne.update();
+    playerOne.render(context);
+    playerTwo.update();
+    playerTwo.render(context);
+    checkCollision(playerOne, playerTwo);
+    checkGameEnd();
+  }
+
+  function gameStateGameOver() {
+    context.fillStyle = BASE.CANVAS_BACKGROUND;
+    context.fillRect(0, 0, BASE.CANVAS_WIDTH, BASE.CANVAS_HEIGHT);
+    // render text
+    context.fillStyle = BASE.CANVAS_TEXTCOLOR;
+    context.font = BASE.CANVAS_FONT;
+    context.textBaseline = 'top';
+    context.fillText (winner + ' wins!!', 50, 90);
+    context.fillText ("Press Enter To Replay", 50, 200);
+    // reset menu
+    currentMenu = null;
+    utils.playerOneChoice = null;
+    utils.playerTwoChoice = null;
+    utils.currentMenuDisplay = "PLAYER 1";
+  }
+
+  // function to switch application states
+  function switchGameState(newState) {
+    switch (newState) {
```

Comment:
I know it may permeate somewhere else in the code, but in the future consider that If in game constructor you set:
```js
GAME_STATE = {
  TITLE = 0,
  INIT= 1,
  PLAY = 2
  GAME_OVER = 3
}
```
here you can:
```js
currentGameStateFunction = GAME[newState];
```